### ISSUE TITLE
Admin User Management - Activate/Deactivate Toggle User not implemented yet

### ISSUE THREAD

**Issue by taitdigital** [taitdigital/aisle-nine-beta#18]

**Comment by taitdigital** [comment]
@jody-tait 

completed both the user activation and permissions work pending deployment.

also implemented the login which should now show and error message 
"Login Failed: your account has been suspended." 

we need to define in the test plans what each user permission can do and then make a ticket for me to go enforce the permissions, heres a rough Idea to start with for a test plan

no permission: (closed beta) -
"coming soon" landing page
login page
public contact page
redirects to the landing page if visiting the registration page with a code

no permission with invite link: (closed beta) -
same as above plus user registration

no permission: (open beta) -
login page
public home page
search and browse recipes
cannot enter cook mode or do anything in a dialog other than share link or sign up
about, terms, legal pages
public contact page

Beta User - 
can use all current app features, 
beta homepage,
can post the the beta message board (when it's made)
no limits on lists or shares

Standard User -
can access all "free tier" features TBD (we need to think about list sizes and share limits)

Moderator User -
can access ingredients, recipes, news, dashboard
cannot access categories, users or system

Admin User -
can access all areas with the following exceptions:
cannot access the system page
cannot change user permissions (when they create a user its a beta_user always until beta ends then its a standard user)

Super Admin User -
can access all areas including system.
can change user permissions

**Comment by jody-tait** [comment]
My initial comments / thoughts:

no permission: (closed beta) -
"coming soon" landing page
login page
public contact page
redirects to the landing page if visiting the registration page with a code

**All good**

no permission with invite link: (closed beta) -
same as above plus user registration

**All good**

no permission: (open beta) -
login page
public home page
search and browse recipes
cannot enter cook mode, or do anything in a dialog other than share link or sign up
about, terms, legal pages
public contact page

**Can filter but those settings aren't saved
Can “heart” recipes, but those settings aren't saved
Can view community page but not comment (depending on what the Community page will entail)
Can view Recommended recipes**

Beta User -
can use all current app features,
beta homepage,
can post the the beta message board (when it's made)
no limits on lists or shares

**All good**

Standard User -
can access all "free tier" features TBD (we need to think about list sizes and share limits)

**Share a shopping list / meal plan / store with up to 3 other people
How often will there be more than 4 in a household that will access to it
Max of 5 shopping lists, and stores
My logic is Pantry / Fridge / Freezer / Deep freeze / other as a “Home User” - if someone were to use it to that extent
Max of 6 meal plans
That’s 4 weeks worth of meal plans + 2 ad hoc**

Moderator User -
can access ingredients, recipes, news, dashboard
cannot access categories, users or system

**All good**

Admin User -
can access all areas with the following exceptions:
cannot access the system page
cannot change user permissions (when they create a user its a beta_user always until beta ends then its a standard user)

**All Good**

Super Admin User -
can access all areas including system.
can change user permissions

**All Good**

**Comment by taitdigital** [comment]
ok, please copy the requirements into a new ticket for me to develop once the test plan is done
you should be able to activate and deactive users now and set the permissions

add a test that right now only admins and super admins can login to admin, we'll update to include moderators later
